src/components/ContractPanel.tsx

```diff
@@ -12,17 +12,26 @@ const ContractPanel: React.FC<Props> = (props) => {
   const [authed, setAuthed] = useState(false);
   const [error, setError] = useState<unknown>();
   const [loading, setLoading] = useState(false);
-
-  const [isMintModalOpen, setIsMintModalOpen] = useState(false);
-  const [isBurnModalOpen, setIsBurnModalOpen] = useState(false);
+  const [owner, setOwner] = useState(false);
 
   const { contract, address } = props;
 
+  useEffect(() => {
+    contract
+      .owner()
+      .then((owner) => {
+        if (owner === address) return setOwner(true);
+      })
+      .catch((err) => setError(err));
+  }, [address]);
+
```

Comment:
fetch the contract owner once, on load. then just `const isOwner = address === owner`

Reply:
👍